tests/src/main/java/com/hazelcast/stabilizer/tests/map/MapEvictAndStoreTest.java

```diff
@@ -0,0 +1,91 @@
+package com.hazelcast.stabilizer.tests.map;
+
+import com.hazelcast.config.MapConfig;
+import com.hazelcast.config.MapStoreConfig;
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.core.IAtomicLong;
+import com.hazelcast.core.IMap;
+import com.hazelcast.logging.ILogger;
+import com.hazelcast.logging.Logger;
+import com.hazelcast.stabilizer.test.TestContext;
+import com.hazelcast.stabilizer.test.annotations.Run;
+import com.hazelcast.stabilizer.test.annotations.Setup;
+import com.hazelcast.stabilizer.test.annotations.Verify;
+import com.hazelcast.stabilizer.test.utils.ThreadSpawner;
+import com.hazelcast.stabilizer.tests.map.helpers.MapStoreWithCounterPerKey;
+
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static com.hazelcast.stabilizer.tests.helpers.HazelcastTestUtils.isMemberNode;
+import static junit.framework.Assert.assertEquals;
+
+public class MapEvictAndStoreTest {
+
+    private final static ILogger log = Logger.getLogger(MapEvictAndStoreTest.class);
+
+    //properties
+    public int threadCount = 10;
+    public String basename = "MapEvictAndStore";
+    public int logInterval = 10;
+    public int maxTestSize = 100000;
+
+    private TestContext testContext;
+    private HazelcastInstance targetInstance;
+    private IAtomicLong keyCounter = null;
+    private int maxMapSize = 5000;
+
+    @Setup
+    public void setup(TestContext testContext) throws Exception {
+        this.testContext = testContext;
+        targetInstance = testContext.getTargetInstance();
+
+        keyCounter = targetInstance.getAtomicLong("keyCounter");
+
+        MapConfig mapConfig = targetInstance.getConfig().getMapConfig(basename);
+        maxMapSize = mapConfig.getMaxSizeConfig().getSize();
+        MapStoreConfig mapStoreConfig = mapConfig.getMapStoreConfig();
+        mapStoreConfig.setWriteCoalescing(true);
+    }
+
+    @Run
+    public void run() {
+        ThreadSpawner spawner = new ThreadSpawner(testContext.getTestId());
+        for (int k = 0; k < threadCount; k++) {
+            spawner.spawn(new Worker());
+        }
+        spawner.awaitCompletion();
+    }
+
+
+    public class Worker implements Runnable {
+        @Override
+        public void run() {
+            IMap<Object, Object> map = targetInstance.getMap(basename);
+            while (!testContext.isStopped() && keyCounter.get()  < maxTestSize) {
```

Comment:
with this test logic,  this test is no longer a run for a time period, which could be set for seconds to day,  but on the 
keyCounter.get()  < maxTestSize  , dominating the run time.